internal/server/auth_handlers.go

```diff
@@ -136,6 +173,34 @@ func (h *auth) Login(c echo.Context) error {
 	return c.JSON(http.StatusOK, login)
 }
 
+func (h *auth) LoginPKCE(c echo.Context) error {
+	// Filter params
+	var params service.LoginParams
+	if err := c.Bind(&params); err != nil {
+		log.Println("Could not get parameters:", err)
+		return c.JSON(http.StatusBadRequest, sferror.New("Could not get credentials."))
+	}
+	params.UserAgent = c.Request().UserAgent()
+	params.Session = currentSession(c)
+
+	if params.Email == "" || params.Password == "" || params.CodeVerifier == "" {
+		return c.JSON(http.StatusUnauthorized, sferror.New("Invalid login credentials."))
+	}
+
+	hash := sha256.Sum256([]byte(params.CodeVerifier))
+	base64_hash := fmt.Sprintf("%x", hash[:])
+	res, err := h.db.RemovePKCE(base64.RawStdEncoding.EncodeToString([]byte(base64_hash)))
```

Comment:
I found to much logic in the database handler (`storm.go`), it should only contains query stuff and not business logic.

How about to create a service to handle the PKCE process?
Which provides:
- Challenge creation and storing
- Computing the challenge from the code_verifier (sha256 stuff)
- Revoke expired challenges, checking if the computed challenge exist in database and then cleaning the database (aka `h.db.RemovePKCE` processing)

Reply:
will do